### PR TITLE
Allow configuring Jackson's TimeZone via configuration

### LINKED ISSUE #15077: timezone setting for jackson in application.properties
**Description**
I would like to set jackson timezone to make Json output displayed with users' timezone. Here request a configuration similar to "spring.jackson.time-zone" in application.properties. It is much convenient than annotation setting like "@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Tokyo")" . As the latter one needs to be set for each date field. Thanks much!

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #15077

### CODE REVIEW

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java [hunk 1]
```diff
@@ -178,6 +179,23 @@ private void registerModuleIfOnClassPath(String moduleClassName,
...
+        if (jacksonBuildTimeConfig.timezone.isPresent()) {
+            /*
+             * We need to make timezone a String instead of a java.util.TimeZone class
+             * because:
+             * 1) TimeZone cannot automatically be handled by the BytecodeRecorder
+             * 2) Handling it would require us to add non-JDK classes (i.e. sun.util.calendar.ZoneInfo) to bytecode recording
+             */
+            String timeZoneStr = jacksonBuildTimeConfig.timezone.get();
+            TimeZone timeZone = TimeZone.getTimeZone(timeZoneStr);
+            if ("GMT".equals(timeZone.getID()) && !timeZoneStr.startsWith("GMT")) {
```
Comment (gsmet) [inline extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java:192]: Open question: I wonder if we should use `ZoneId` instead and only convert it to `TimeZone` when pushing it to Jackson. It seems to have a saner behavior when parsing (and there is a `checkAvailable` boolean that might make sense to use).

**Review by gsmet — COMMENTED** [review]
Added a question inline.

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java:192]: That's interesting. Let me explore

Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: I pushed a second commit that introduces support for `ZoneId`

File: integration-tests/container-image/quarkus-standard-way/src/test/java/io/quarkus/it/container/image/GreetingResourceIT.java [hunk 2]
```diff
@@ -3,12 +3,11 @@
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.CoreMatchers.is;
 
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import io.quarkus.test.junit.QuarkusIntegrationTest;
 
-@Disabled("This test seems flaky")
+//@Disabled("This test seems flaky")
```
Comment (gsmet) [inline integration-tests/container-image/quarkus-standard-way/src/test/java/io/quarkus/it/container/image/GreetingResourceIT.java:10]: Hmm?

Comment (geoand) [reply integration-tests/container-image/quarkus-standard-way/src/test/java/io/quarkus/it/container/image/GreetingResourceIT.java:10]: Ah damn, will revert

Comment (geoand) [reply integration-tests/container-image/quarkus-standard-way/src/test/java/io/quarkus/it/container/image/GreetingResourceIT.java:10]: Fixed

**Review by gsmet — APPROVED** [review]
Looks cool now and paves the way for future timezone settings.

**Comment by geoand** [comment]
Yeah, your idea about `ZoneId` really made the difference!

**Comment by gsmet** [comment]
Something is not right:
```
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:315)
	... 15 more
Caused by: java.lang.NoClassDefFoundError: io/quarkus/deployment/recording/substitutions/ZoneIdSubstitution
	at io.quarkus.deployment.steps.JacksonProcessor$pushConfigurationBean290634148.deploy_0(JacksonProcessor$pushConfigurationBean290634148.zig:62)
	at io.quarkus.deployment.steps.JacksonProcessor$pushConfigurationBean290634148.deploy(JacksonProcessor$pushConfigurationBean290634148.zig:40)
	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:163)
	... 15 more
The output of the Quarkus build can be found at /home/runner/work/quarkus/quarkus/integration-tests/hibernate-orm-panache/target/quarkus-prod-mode-test438526196068243597/build-result
Caused by: java.lang.ClassNotFoundException: io.quarkus.deployment.recording.substitutions.ZoneIdSubstitution
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:104)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:60)
	... 18 more
```

**Comment by geoand** [comment]
Ouch...

I'll have to take a closer look

**Comment by geoand** [comment]
Should be good now

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/ZoneIdConverter.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.io.Serializable;
+import java.time.ZoneId;
+
+import javax.annotation.Priority;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+/**
+ * A converter to support locales.
```
Comment (machi1990) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/ZoneIdConverter.java:13]: A converter to support zoneId?

Comment (geoand) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/ZoneIdConverter.java]: Good catch - fixed

**Review by machi1990 — APPROVED** [review]

**Pull request merged**